validation/src/main/java/io/micronaut/validation/validator/constraints/ConstraintValidatorContext.java

```diff
@@ -48,5 +48,23 @@
      * @return The root bean under validation.
      */
     @Nullable Object getRootBean();
+
+    /**
+     * Sets a message template to be used for the validation error to override default.
+     *
+     * @param messageTemplateOverride the message to override default
+     */
+    default void overrideMessageTemplate(@Nullable final String messageTemplateOverride) {
+        throw new UnsupportedOperationException("not implemented");
+    }
+
+    /**
+     * Get the message template override.
+     *
+     * @return the message override
+     */
+    @Nullable default String getMessageTemplateOverride() {
```

Comment:
Does this need to be `public` API? Seems it is only used internally. Either way it should return `Optional` and if it must be public be annotated with `@since 2.5.0`